### PR TITLE
Fix Resources

### PR DESCRIPTION
**Pull request by MarisaGoergen**
- Fix using Fullname to organizie resource types
- Fix registering the ResourceTypeTreeFacade
- Fix invoking methods of resources with void as return value
- Trying to access a resource that doesn't exists stops throwing an exception
- When the AssemblyResource creates an AssemblyFoo resource during the creation, the id will be updated

### CODE REVIEW

File: src/Moryx.Resources.Samples/AssembleResource.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ protected override void OnInitialize()
             if (Setup == null)
             {
                 Setup = Graph.Instantiate<AssembleFoo>();
+                Graph.Save(Setup);
```
Comment (Toxantron) [inline src/Moryx.Resources.Samples/AssembleResource.cs:32]: That should not be necessary. Call Changed **must** take care of this.

File: src/Moryx.Resources.Management/Facades/ResourceManagementFacade.cs [hunk 2]
```diff
@@ -131,8 +131,6 @@ public TResult Read<TResult>(long id, Func<Resource, TResult> accessor)
             ValidateHealthState();
 
             var resource = ResourceGraph.Get(id);
-            if (resource == null)
-                throw new KeyNotFoundException($"No resource with Id {id} found!");
```
Comment (Toxantron) [inline src/Moryx.Resources.Management/Facades/ResourceManagementFacade.cs:135]: Is this still consistent with the behavior of Get(long id)?

File: src/Moryx.Resources.Samples/AssembleResource.cs (hunk 1 above)
Comment (MarisaGoergen) [reply src/Moryx.Resources.Samples/AssembleResource.cs]: I've fixed that in the ResourceLinker

File: src/Moryx.Resources.Management/Facades/ResourceManagementFacade.cs (hunk 2 above)
Comment (MarisaGoergen) [reply src/Moryx.Resources.Management/Facades/ResourceManagementFacade.cs:135]: The Graph also returns null and the Facade shoudn't change that

**Review by Toxantron — APPROVED** [review]

**Pull request merged**